Plugins/GeoModel/src/GeoModelToDetVol.cpp

```diff
@@ -0,0 +1,197 @@
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrapezoidVolumeBounds.hpp"
+#include "Acts/Navigation/InternalNavigation.hpp"
+
+#include <GeoModelKernel/GeoBox.h>
+#include <GeoModelKernel/GeoPcon.h>
+#include <GeoModelKernel/GeoShapeShift.h>
+#include <GeoModelKernel/GeoShapeSubtraction.h>
+#include <GeoModelKernel/GeoShapeUnion.h>
+#include <GeoModelKernel/GeoSimplePolygonBrep.h>
+#include <GeoModelKernel/GeoTrd.h>
+#include <GeoModelKernel/GeoTube.h>
+#include <GeoModelKernel/GeoTubs.h>
+
+namespace Acts {
+namespace GeoModel {
+std::shared_ptr<Experimental::DetectorVolume> convertVolume(
+    const GeometryContext& context, const GeoShape& shape,
+    const std::string& name, const Transform3& transform) {
+  auto portalGenerator = Experimental::defaultPortalAndSubPortalGenerator();
+  if (shape.typeID() == GeoTube::getClassTypeID()) {
+    const GeoTube& tube = static_cast<const GeoTube&>(shape);
+    std::shared_ptr<CylinderVolumeBounds> bounds =
+        std::make_shared<CylinderVolumeBounds>(tube.getRMin(), tube.getRMax(),
+                                               tube.getZHalfLength());
+    return Experimental::DetectorVolumeFactory::construct(
+        portalGenerator, context, name, transform, bounds,
+        Experimental::tryAllPortalsAndSurfaces());
+  } else if (shape.typeID() == GeoTubs::getClassTypeID()) {
+    const GeoTubs& tubs = static_cast<const GeoTubs&>(shape);
+    std::shared_ptr<CylinderVolumeBounds> bounds =
+        std::make_shared<CylinderVolumeBounds>(tubs.getRMin(), tubs.getRMax(),
+                                               tubs.getZHalfLength(),
+                                               tubs.getDPhi() / 2);
+    Acts::Transform3 newTransform = Acts::Transform3::Identity();
+    newTransform.translate(transform.translation());
+    newTransform.rotate(Eigen::AngleAxisd(tubs.getSPhi() + 0.5 * tubs.getDPhi(),
+                                          Acts::Vector3::UnitZ()));
+    return Experimental::DetectorVolumeFactory::construct(
+        portalGenerator, context, name, newTransform, bounds,
+        Experimental::tryAllPortalsAndSurfaces());
+  } else if (shape.typeID() == GeoBox::getClassTypeID()) {
+    const GeoBox& box = static_cast<const GeoBox&>(shape);
+    std::shared_ptr<CuboidVolumeBounds> bounds =
+        std::make_shared<CuboidVolumeBounds>(
+            box.getXHalfLength(), box.getYHalfLength(), box.getZHalfLength());
+    return Experimental::DetectorVolumeFactory::construct(
+        portalGenerator, context, name, transform, bounds,
+        Experimental::tryAllPortalsAndSurfaces());
+  } else if (shape.typeID() == GeoSimplePolygonBrep::getClassTypeID()) {
+    // Will change this in the future
+    double xmin{0};
+    double xmax{0};
+    double ymin{0};
+    double ymax{0};
+    double zmin{0};
+    double zmax{0};
+    const GeoSimplePolygonBrep& brep =
```

Comment:
Maybe a trapezoid is a better approximation? https://gitlab.cern.ch/GeoModelDev/GeoModel/-/blob/main/GeoModelCore/GeoModelHelpers/src/GeoShapeUtils.cxx#L152-153. Would give you the edge points of the brep and you could just sort them by plus and minus z. Or what ever.

Reply:
I tried doing this but I think for now it will be best to just leave it as a simple bounding box and then re-evaluating later if needed.